### PR TITLE
[SDXL] Make sure multi batch prompt embeds works

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

Currently the following does not work: 

```py
import hf_image_uploader as hiu
import torch
from compel import Compel, ReturnedEmbeddingsType
from diffusers import DiffusionPipeline

pipe = DiffusionPipeline.from_pretrained(
  "stabilityai/stable-diffusion-xl-base-1.0",
  variant="fp16",
  torch_dtype=torch.float16
)
pipe2 = DiffusionPipeline.from_pretrained(
  "stabilityai/stable-diffusion-xl-refiner-1.0",
  variant="fp16",
  torch_dtype=torch.float16
)
pipe.enable_model_cpu_offload()
pipe2.enable_model_cpu_offload()

compel = Compel(
  tokenizer=[pipe.tokenizer, pipe.tokenizer_2] ,
  text_encoder=[pipe.text_encoder, pipe.text_encoder_2],
  returned_embeddings_type=ReturnedEmbeddingsType.PENULTIMATE_HIDDEN_STATES_NON_NORMALIZED,
  requires_pooled=[False, True]
)

compel2 = Compel(
  tokenizer=pipe.tokenizer_2,
  text_encoder=pipe.text_encoder_2,
  returned_embeddings_type=ReturnedEmbeddingsType.PENULTIMATE_HIDDEN_STATES_NON_NORMALIZED,
  requires_pooled=True,
)

# apply weights
prompt = ["a red cat playing with a (ball)1.5", "a red cat playing with a (ball)0.6"]
conditioning, pooled = compel(prompt)
conditioning2, pooled2 = compel2(prompt)

# generate image

for _ in range(3):
    generator = [torch.Generator().manual_seed(i) for i in range(len(prompt))]

    image = pipe(prompt_embeds=conditioning, pooled_prompt_embeds=pooled, generator=generator, num_inference_steps=30, output_type="latent").images
    image = pipe2(image=image, prompt_embeds=conditioning2, pooled_prompt_embeds=pooled2, generator=generator, num_inference_steps=20).images[0]
    hiu.upload(image, "patrickvonplaten/images")
```

This PR fixes it.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [ ] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: tests/pipelines/stable_diffusion_xl/test_stable_diffusion_xl.py [hunk 1]
```diff
@@ -261,6 +261,43 @@ def test_stable_diffusion_xl_offloads(self):
         assert np.abs(image_slices[0] - image_slices[1]).max() < 1e-3
         assert np.abs(image_slices[0] - image_slices[2]).max() < 1e-3
 
+    def test_stable_diffusion_xl_img2img_prompt_embeds_only(self):
```
Comment (DN6) [inline tests/pipelines/stable_diffusion_xl/test_stable_diffusion_xl.py:264]: For completeness, do we want to test the prompt embed case where prompt is passed in as a string? Currently all tests related to prompt embeds pass in the prompts as a list.

**Review by DN6 — APPROVED** [review]
LGTM 👍🏽 . Just had one small comment/question.

File: tests/pipelines/stable_diffusion_xl/test_stable_diffusion_xl.py [hunk 2]
```diff
@@ -261,6 +261,43 @@ def test_stable_diffusion_xl_offloads(self):
...
+        components = self.get_dummy_components()
+        sd_pipe = StableDiffusionXLPipeline(**components)
+        sd_pipe = sd_pipe.to(torch_device)
+        sd_pipe = sd_pipe.to(torch_device)
```
Comment (patrickvonplaten) [inline tests/pipelines/stable_diffusion_xl/test_stable_diffusion_xl.py:268]:
```suggestion
```

File: tests/pipelines/stable_diffusion_xl/test_stable_diffusion_xl.py (hunk 1 above)
Comment (patrickvonplaten) [reply tests/pipelines/stable_diffusion_xl/test_stable_diffusion_xl.py:264]: Hmm we always normalize single strings to a list in the code, so maybe not necessary here?

File: tests/pipelines/stable_diffusion_xl/test_stable_diffusion_xl_img2img.py [hunk 3]
```diff
@@ -559,6 +559,43 @@ def test_stable_diffusion_xl_img2img_negative_prompt_embeds(self):
         # make sure that it's equal
         assert np.abs(image_slice_1.flatten() - image_slice_2.flatten()).max() < 1e-4
 
+    def test_stable_diffusion_xl_img2img_prompt_embeds_only(self):
+        components = self.get_dummy_components()
+        sd_pipe = StableDiffusionXLImg2ImgPipeline(**components)
+        sd_pipe = sd_pipe.to(torch_device)
+        sd_pipe = sd_pipe.to(torch_device)
```
Comment (patrickvonplaten) [inline tests/pipelines/stable_diffusion_xl/test_stable_diffusion_xl_img2img.py:566]:
```suggestion
```

**Comment by bghira** [comment]
this fixes the issue i described a while back, thank you
